akka-stream-tests/src/test/java/akka/stream/javadsl/FlowTest.java

```diff
@@ -192,6 +193,26 @@ public void mustBeAbleToUseStatefulMap() throws Exception {
     Assert.assertEquals("[1, 2][3, 4][5]", grouped.toCompletableFuture().get(3, TimeUnit.SECONDS));
   }
 
+  @Test
+  public void mustBeAbleToUseMapWithResource() {
+    final AtomicBoolean gate = new AtomicBoolean(true);
```

Comment:
This atomic is not needed, we verify that it emits "end" and it can't unless it executes the close-lambda. Same below.